File: leetcode/50-pow/main.go
package main

import "fmt"

func myPow(x float64, n int) float64 {
	if n < 0 {
		x, n = 1/x, -n
	}
	result := 1.0
	for n > 0 {
		if n&1 == 1 {
			result *= x
		}
		x, n = x*x, n>>1
	}
	return result
}

func main() {
	fmt.Println(myPow(2.0, -2))
}

File: leetcode/50-pow/main.py
class Solution:
    def myPow(self, x: float, n: int) -> float:
        result = 1.0
        if n < 0:
            x, n = 1.0 / x, -n
        while n:
            if n & 1:
                result *= x
            x, n = x * x, n >> 1
        return result


solution = Solution()
print(solution.myPow(2, -2))
